jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java

```diff
@@ -71,82 +94,31 @@ public void onResponse(Stream.Client stream, HeadersFrame frame)
         MetaData.Response response = (MetaData.Response)frame.getMetaData();
         httpResponse.version(response.getHttpVersion()).status(response.getStatus()).reason(response.getReason());
 
-        if (responseBegin(exchange))
+        responseBegin(exchange);
+        HttpFields headers = response.getFields();
+        for (HttpField header : headers)
         {
-            HttpFields headers = response.getFields();
-            for (HttpField header : headers)
-            {
-                if (!responseHeader(exchange, header))
-                    return;
-            }
-
-            // TODO: add support for HttpMethod.CONNECT.
-
-            if (responseHeaders(exchange))
-            {
-                int status = response.getStatus();
-                if (frame.isLast() || HttpStatus.isInterim(status))
-                    responseSuccess(exchange);
-                else
-                    stream.demand();
-            }
-            else
-            {
-                if (LOG.isDebugEnabled())
-                    LOG.debug("stalling response processing, no demand after headers on {}", this);
-            }
+            responseHeader(exchange, header);
```

Comment:
Check for `exchange.isResponseComplete()` here.

Reply:
Done.